content/articles/building-faq-page-using-react-styled-components-and-react/index.md

```diff
@@ -0,0 +1,290 @@
+user-select: none;
+
+@media (max-width: 600px) {
+font-size: 16px;
+line-height: 22px;
+}
+`;
+export const Title = styled.h1`
+font-size: 50px;
+line-height: 6;
+margin-top: 0;
+margin-bottom: 8px;
+color: black;
+text-align: center;
+
+@media (max-width: 600px) {
+font-size: 35px;
+}
+
+color: black;
+`;
+
+`;
+```
+
+Our styles functions are fairly easy to understand. For starters those are not pure CSS codes, we are simply using the Styled-component library to implement JavaScript looking like CSS (CSS-in-JS), which is the beauty of this approach. The first thing to do when using styled-components libraries is to import it to any file you wish to use it in as shown above.
+Finally, we added some responsiveness to keep our page looking good on large, medium, and small screens.
+
+#### Step 6: Mapping and displaying the FAQs (Faq.js)
+It is believed that the most preferred way of displaying items from an array is with the use of a `map` function. In our `App.js`, first, we clear the default boilerplate, thereafter We will import our dummies question from the JSON file we created earlier, then set up the `map` function by:
+
+```JavaScript
+import React from "react";
+import faqsData from "./Faq.json";
+import Accordion from "./Accordion";
+export function App() {
+return (
+<Accordion>
+<Accordion.Title>Frequently Asked Questions</Accordion.Title>
+{faqsData.map((item) => (
+<Accordion.Item key={item.id}>
+<Accordion.Question>{item.question}</Accordion.Question>
+<Accordion.Body>{item.answers}</Accordion.Body>
+</Accordion.Item>
+))}
+<h5 >
+Question not on the list? Contact out help desk for further enquiries
+</h5>
+</AppForm>
+</Accordion>
+);
+}
+```
+from the snippet above, we imported our faq, mapped, and displayed them with their corresponding answers. we also made use of our reusable `Accordion` component.
+Our FAQ page is ready for viewing on our browser. To do that, we open the `command terminal` in our text editor and run the command below
+
+```bash
+npm start
+```
+
+Alternatively
+
+```bash
+yarn start
+```
+
+Once the development server is up and running, you should get a page that looks like this:
+![faq page](/engineering education/ building-faq-page-using-react-styled-components-with-react/image1)
+Finally, to further enhance the beauty of our page, we will go ahead and add some font classes, remove unnecessary margins and change the background color of the page to black. To do that, delete the `app.css` file, thereafter, in the `index.css` file, simply paste the code snippet below:
+
+```CSS
+body {
+background-color: black !important;
+margin: 0;
+color: white;
+font-family: 'Arial', 'Helvetica Neue', 'sans-serif',;
+-webkit-font-smoothing: antialiased;
+-moz-osx-font-smothing: grayscale;
+font-size: 16px;
+}
+```
+
+Once the code snippet is correctly implemented, your FAQ page should look like this below:
+![faq page](/engineering education/ building-faq-page-using-react-styled-components-with-react/image2)

```

Comment:
Add extension.

The image path is not right. Please check the guidelines again or refer other articles.